cmd/docker/docker.go

```diff
@@ -240,16 +241,17 @@ func tryPluginRun(dockerCli command.Cli, cmd *cobra.Command, subcommand string,
 	// we send a SIGKILL to the plugin process and exit
 	go func() {
 		retries := 0
-		for s := range signals {
+		for range signals {
+			if dockerCli.Out().IsTerminal() {
```

Comment:
I'm not sure if `dockerCli.Out()` is ideal here, eg. for a command like `build -o type=tar . > t.tar` . Although afaics, the actual behavior change from this line is minimal.